### PR TITLE
Allow overriding selectors and names

### PR DESCRIPTION
**Pull request by alex-berger**
#### Why is this pull request needed and what does it do?

This allows adapting to non-standard setups like for example AWS EKS CoreDNS Add-on, which use different labels, selectors and object names. Specifically, it allows 

- overriding of selectors for _Deployment_, _Services_, _PodDisruptionBudget_ and _ServiceMonitor_.
- overriding the _ClusterRole_ and _ClusterRoleBinding_ names.

This is achieved by introducing the following optional new settings:

| Parameter                                      | Description                                                                                                                               | Default                                                      |
| :--------------------------------------------- | :---------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------------------------- |
| `service.selector`                             | Pod selector                                                                                                                              | `{}`                                                         |
| `deployment.selector`                          | Pod selector                                                                                                                              | `{}`                                                         |
| `prometheus.service.selector`                  | Pod selector                                                                                                                              | `{}`                                                         |
| `prometheus.monitor.selector`                  | Service selector                                                                                                                          | `{}`                                                         |
| `clusterRole.nameOverride`                     | ClusterRole name override                                                                                                                 |                                                              |

The changes are fully backward compatible and one has to explicitly opt-in to override the default behavior. 

#### Which issues (if any) are related?

- https://github.com/coredns/helm/issues/97

Checklist:

* [X] I have bumped the chart version according to [versioning](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#versioning).
* [X] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have signed off all my commits as required by [DCO](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#developer-certificate-of-origin).

Changes are automatically published when merged to `main`. They are not published on branches.

<details>
  <summary>Note on DCO</summary>

  If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

### CODE REVIEW

**Comment by joshuabaird** [comment]
This would be very helpful for users who are trying to install the helm chart over a "stock" EKS deployment or even a deployment where the managed EKS addon for coredns is being used.

**Comment by alex-berger** [comment]
Randomly notifying one of the maintainers for review @mrueg, no urgency or pressure, just would love to make progress on this 😇.

File: charts/coredns/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: coredns
-version: 1.30.0
+version: 1.30.1
```
Comment (hagaibarel) [inline charts/coredns/Chart.yaml:3]: Let's up the minor version, we're adding stuff not just fixing

```suggestion
version: 1.31.0
```

**Review by hagaibarel — CHANGES_REQUESTED** [review]
Small nit, otherwise LGTM

**Comment by hagaibarel** [comment]
Thanks @alex-berger !

**Comment by joshuabaird** [comment]
Thanks, @alex-berger!

**Pull request merged**